### PR TITLE
V0.6.4

### PR DESCRIPTION
**Pull request by That3Percent**
New APIs: `init_with`, `flatten`, `prefer`, and `idle`

### CODE REVIEW

File: src/combinators/mod.rs [hunk 1]
```diff
@@ -406,3 +406,115 @@ where
...
+            select! {
+                biased;
+
+                one = source_1.next() => {
+                    if let Ok(one) = one {
+                        writer.write(one);
+                        break;
+                    } else {
+                        loop {
+                            writer.write(source_2.next().await?);
```
Comment (Theodus) [inline src/combinators/mod.rs:445]: Shouldn't we `drop(source_1)` here?

**Review by Theodus — APPROVED** [review]
LGTM

File: src/combinators/mod.rs (hunk 1 above)
Comment (That3Percent) [reply src/combinators/mod.rs:445]: Correct me if I'm wrong, but I don't think that accomplishes anything.

The reason for `drop(source_2)` on line 458 is to allow writes to cancel. On line 445 `source_1` has already returned `Err(Closed)` so this concern does not apply.

Comment (Theodus) [reply src/combinators/mod.rs:445]: Ok, I see. The drop below is for the break on line 442, not the one on line 453. I understand now.

**Pull request merged**